sql/src/main/java/io/crate/protocols/postgres/types/PGTypes.java

```diff
@@ -105,15 +105,21 @@ public static PGType get(DataType type) {
             DataType<?> innerType = ((CollectionType) type).innerType();
             if (innerType instanceof CollectionType) {
                 // if this is a nested collection stream it as JSON because
-                // postgres binary format doesn't support multidimensional arrays with sub-arrays of different length
+                // postgres binary format doesn't support multidimensional arrays
+                // with sub-arrays of different length
                 // (something like [ [1, 2], [3] ] is not supported)
                 return JsonType.INSTANCE;
+            } else if (innerType.id() == ObjectType.ID) {
+                return PGArray.JSON_ARRAY;
```

Comment:
Isn't this the part for which we added a breaking changes entry in master?

Reply:
no, it is needed to resolve a pg type that corresponds to crate object or object array data type such as we implemented `hashCode` and `equals` in `ObjectType` and cannot use, for example, an instance of object data type with inner types to look up a pg type mapping in `CRATE_TO_PG_TYPES`. 

These commits do not contain any breaking changes. The inner cast function signatures were changed, but it is still possible to resolve cast functions for the former signatures.

Reply:
ah, but this line is wrong, it should be mapped to `PGArray.JSON` https://github.com/crate/crate/blob/8d568694fbc30fd58ec4ad5114729a3ff6577d06/sql/src/main/java/io/crate/protocols/postgres/types/PGTypes.java#L69